### PR TITLE
[skip ci] Add Cassandra SSL documentation

### PR DESCRIPTION
**Pull request by SauronSauron**
Only documentation for self signed certificate

### CODE REVIEW

File: doc/Advanced-configuration.md [hunk 1]
```diff
@@ -398,6 +398,61 @@ If they are not present - defaults are used (connection to `localhost:9042` with
  ]}.
 ```
 
+#### SSL connection setup
+
+In order to establish a secure connection to Cassandra you must make some changes in the MongooseIM and Cassandra configuration files.
+
+##### Create MongooseIM keyspace
```
Comment (fenek) [inline doc/Advanced-configuration.md]: This section should go to https://github.com/esl/MongooseIM/blob/master/doc/advanced-configuration/database-backends-configuration.md

File: doc/Advanced-configuration.md [hunk 2]
```diff
@@ -398,6 +398,61 @@ If they are not present - defaults are used (connection to `localhost:9042` with
...
+    keystore_password: your_password
+```
+Save the changes and restart Cassandra.
+
+##### Enable MongooseIM to connect with SSL
+Find `cassandra_servers` in `ejabberd.cfg` and add the following line:
+```
+{cassandra_servers, [{default, [{ssl, [{verify, verify_none}]}]}]}.
+```
+Save the changes and rebuild MongooseIM.
```
Comment (fenek) [inline doc/Advanced-configuration.md]: Restart, not rebuild.

File: doc/Advanced-configuration.md [hunk 3]
```diff
@@ -398,6 +398,54 @@ If they are not present - defaults are used (connection to `localhost:9042` with
...
+    enabled: true
+    keystore: /your_certificate_directory/server.keystore
+    keystore_password: your_password
+```
+Save the changes and restart Cassandra.
+
+##### Enable MongooseIM to connect with SSL
+Find `cassandra_servers` in `ejabberd.cfg` and add the following line:
+```
+{cassandra_servers, [{default, [{ssl, [{verify, verify_none}]}]}]}.
```
Comment (DenysGonchar) [inline doc/Advanced-configuration.md]: how about to describe the **verify_peer** option as well?

Comment (fenek) [reply doc/Advanced-configuration.md]: :+1: Good idea!

File: doc/Advanced-configuration.md [hunk 4]
```diff
@@ -398,6 +398,68 @@ If they are not present - defaults are used (connection to `localhost:9042` with
...
+
+###### CA-signed certificate
+
+Find `cassandra_servers` in `ejabberd.cfg` and add the following line:
+```
+{cassandra_servers, [{default, [{ssl, [{cacertfile,
+                                        "$CRTS/rootCA.pem"},
+                                        {verify, verify_peer}]}]}]}.
+```
+Where `$CRTS` is the path to the directory containing the certificate. Save the changes and restart MongooseIM.
```
Comment (DenysGonchar) [inline doc/Advanced-configuration.md]: i must say that usage of environment variable inside the erlang config file can be confusing and misinterpreted. you can not use environment variables here!

better use some fake path, like "/path/to/rootCA.pem"

**Pull request merged**